src/main/java/com/theelixrlabs/healthcare/service/PatchPatientService.java

```diff
@@ -0,0 +1,72 @@
+package com.theelixrlabs.healthcare.service;
+
+import com.theelixrlabs.healthcare.constants.MessageConstants;
+import com.theelixrlabs.healthcare.constants.PatientConstants;
+import com.theelixrlabs.healthcare.dto.PatientDto;
+import com.theelixrlabs.healthcare.exceptionHandler.CustomException;
+import com.theelixrlabs.healthcare.exceptionHandler.ResourceNotFoundException;
+import com.theelixrlabs.healthcare.model.PatientModel;
+import com.theelixrlabs.healthcare.repository.DoctorRepository;
+import com.theelixrlabs.healthcare.repository.PatientRepository;
+import com.theelixrlabs.healthcare.utility.MessageUtil;
+import com.theelixrlabs.healthcare.validation.Validator;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class PatchPatientService {
+    private final PatientRepository patientRepository;
+    private final Validator validator;
+    private final MessageUtil messageUtil;
+    private final DoctorRepository doctorRepository;
+
+    public PatchPatientService(PatientRepository patientRepository, Validator validator, MessageUtil messageUtil, DoctorRepository doctorRepository) {
+        this.patientRepository = patientRepository;
+        this.validator = validator;
+        this.messageUtil = messageUtil;
+        this.doctorRepository = doctorRepository;
+    }
+
+    /**
+     * Replace existing patient details based on request.
+     *
+     * @param patientId  Patient unique UUID
+     * @param patientDTO Data transfer object containing patient information.
+     * @return The replaced patient details.
+     * @throws CustomException If validation fails or patient already exists.
+     */
+    public PatientDto patchPatientById(UUID patientId, PatientDto patientDTO) throws CustomException {
+        validator.validatePatchPatient(patientDTO);
+        Optional<PatientModel> optionalDoctor = patientRepository.findById(patientId);
+        if (optionalDoctor.isPresent()) {
+            PatientModel existingPatient = optionalDoctor.get();
+            if (patientDTO.getPatientFirstName() != null && !patientDTO.getPatientFirstName().equals(existingPatient.getPatientFirstName())) {
```

Comment:
Validator is already doing the empty checks. Do we need this redundant null check? Also do we need to check equals for all fields individually?